e2e_test/batch/functions/cast.slt.part

```diff
@@ -0,0 +1,40 @@
+statement ok
+create table dt (a date, b varchar);
+
+statement ok
+insert into dt values (date('2020-01-23'), '1990-12-19');
+
+query TT
+select
+    date(b)
+from
+dt
+except
+select
+    cast(b as date)
+from dt
+except
+select
+    b::date
+from dt;
+----
+
+statement ok
+drop table dt;
+
+query T
+select date('2030-03-30');
+----
+2030-03-30
+
+query error
+select date('2000-13-03');
+
+query error
+select date('00-00-33');
+
+query error cannot cast type "integer" to "date" in Explicit context
+select date(1);
+
+query error unexpected arguments number
+select date();
```

Comment:
error reported by pg here is: `Query Error: error: function date() does not exist`